### DISCUSSION TITLE
Crash on generating a skeletalmesh

### DISCUSSION THREAD

**Discussion by MarcelEissens** [AndreaCatania/RuntimeSkeletalMeshGenerator#21] [Q&A]
Hi!

For a file-convertion tool I'm working on, I needed to convert a mesh with bones, to something I can export later.
This plugin seems the way to do it, and I have seen a working example (from Fransferdy). However, when I try it with my own mesh, the editor crashes (packaged game doesn't produce a mesh, but doesn't crash).

Using Unreal Engine 5.1 (5.2 didn't work, even with the working example from Fransferdy)
The crash I'm getting is:

Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0xffffffffffffffff

UnrealEditor_Core!_mi_malloc_generic() [D:\build\++UE5\Sync\Engine\Source\ThirdParty\mimalloc\src\page.c:875]
UnrealEditor_Core!_mi_heap_malloc_zero() [D:\build\++UE5\Sync\Engine\Source\ThirdParty\mimalloc\src\alloc.c:148]
UnrealEditor_Core!mi_heap_malloc_zero_aligned_at() [D:\build\++UE5\Sync\Engine\Source\ThirdParty\mimalloc\src\alloc-aligned.c:52]
UnrealEditor_Core!mi_realloc_aligned() [D:\build\++UE5\Sync\Engine\Source\ThirdParty\mimalloc\src\alloc-aligned.c:187]
UnrealEditor_Core!FMallocMimalloc::Realloc() [D:\build\++UE5\Sync\Engine\Source\Runtime\Core\Private\HAL\MallocMimalloc.cpp:133]
UnrealEditor_Core!FMemory::Realloc() [D:\build\++UE5\Sync\Engine\Source\Runtime\Core\Public\HAL\FMemory.inl:56]
UnrealEditor_Core!FMemoryImageAllocatorBase::ResizeAllocation() [D:\build\++UE5\Sync\Engine\Source\Runtime\Core\Private\Containers\MemoryImageAllocator.cpp:50]
UnrealEditor_Engine!TArray<FPositionVertex,TMemoryImageAllocator<0> >::ResizeTo() [D:\build\++UE5\Sync\Engine\Source\Runtime\Core\Public\Containers\Array.h:3047]
UnrealEditor_Engine!TStaticMeshVertexData<FPositionVertex>::ResizeBuffer() [D:\build\++UE5\Sync\Engine\Source\Runtime\Engine\Public\StaticMeshVertexData.h:43]
UnrealEditor_Engine!FPositionVertexBuffer::Init() [D:\build\++UE5\Sync\Engine\Source\Runtime\Engine\Private\PositionVertexBuffer.cpp:107]
UnrealEditor_RuntimeSkeletalMeshGenerator_0013!FRuntimeSkeletalMeshGenerator::GenerateSkeletalMesh() [D:\Downloads\SkeletalTest\Plugins\RuntimeSkeletalMeshGenerator\Source\RuntimeSkeletalMeshGenerator\RuntimeSkeletalMeshGenerator.cpp:426]

I have had it happen that it doesn't crash, which leads to me believing it's something with threads/timing.

If anyone could help me out, that'd be a lifesaver!

Thanks in advance, 
Marcel

**Comment by matiasgql** [comment]
I have the same crash too in 5.2 I think it is a problem because we need to update some types of variables in UE5, like FVector3 to FVector3f, I think it is a memory allocation problem with the two types, if I fix or find other thing I'm gonna tell you.

**Comment by Fransferdy** [comment]
You guys can try using my fork, it has some bugfixes:
https://github.com/Fransferdy/RuntimeSkeletalMeshGenerator
( I think it is ported to UE 5.1, it has been a while, I don't remember very well)

**Reply by matiasgql** [reply]
Well, that's the version I'm using, I use your fork but it doesn't work in UE5.2 (it crashes) and in UE5.3(doesn't crash but doesn't nothing) I also update the code to SkinnedMesh and UE5.3 other changes but still not working
